offlineimap/folder/Base.py

```diff
@@ -863,9 +863,10 @@ def copymessageto(self, uid, dstfolder, statusfolder, register=1):
                 self.deletemessage(uid)
             else:
                 raise OfflineImapError("Trying to save msg (uid %d) on folder "
-                    "%s returned invalid uid %d"% (uid, dstfolder.getvisiblename(),
-                    new_uid), OfflineImapError.ERROR.MESSAGE)
         except (KeyboardInterrupt): # Bubble up CTRL-C.
+                                       "%s returned invalid uid %d" % (uid, dstfolder.getvisiblename(),
+                                                                       new_uid), OfflineImapError.ERROR.MESSAGE)
+            return new_uid
```

Comment:
Looks like incorrect changes.

Reply:
This is incorrect.
At the very least, you need to move lines 867-868 back to after line 865,
Then decide where line 869 should go.
Have you tried running your code on a test mailbox?